### ISSUE TITLE
GCC 4.9 broken

### ISSUE THREAD

**Issue by Unitrunker** [cnjinhao/nana#289]
gcc 4.9 has been broken for at least two months.

More specifically, the travis CI rules for building against gcc 4.9 are broken. 

Is there still demand for .. .say ... Ubuntu 14.04?

**Comment by cnjinhao** [comment]
GCC 4.9 is OK, the problem of the travis CI is that one of test examples is failed to compile because of linking agains to boost_filesystem.

The cmake outputs:
```
-- Found the following Boost libraries:
--   filesystem
--   system

NANA_LINKS  =  -lX11 -lfontconfig -lXft -static-libgcc -static-libstdc++  -pthread -ljpeg -lboost_filesystem -lboost_system
```

But it is still get linker errors:
```
[ 35%] Linking CXX executable file_explorer
CMakeFiles/file_explorer.dir/file_explorer.cpp.o: In function `operator()':
/home/travis/build/cnjinhao/nana-demo/file_explorer.cpp:71: undefined reference to `nana::experimental::filesystem::v1::directory_entry::path() const'
```

nana::experimental::filesystem is using boost::filesystem. Maybe the problem is caused by the libboost-filesystem-dev which is built with other compiler options.

**Comment by qPCR4vir** [comment]
I'm working on that....
but honestly, only to make it work and to learn. It is time to abandon gcc 4.9 and boost::filesystem. They are not c++11.

**Comment by cnjinhao** [comment]
Yes, we should remove support of GCC 4.9 from travis CI of `nana-demo`

**Comment by qPCR4vir** [comment]
fixed, both develop #292  and hotfix #293

**Comment by qPCR4vir** [comment]
See https://github.com/qPCR4vir/nana-demo/pull/13
